### PR TITLE
Audio - Version check 

### PR DESCRIPTION
**Pull request by LakshmiKumar23**
- Checks for vx_rpp version from MIVisionX
- Merge [MIVisionX PR1353](https://github.com/ROCm/MIVisionX/pull/1353) before this
- remove existing rocm version check for the same

### CODE REVIEW

**Review by kiritigowda — DISMISSED** [review]

**Comment by kiritigowda** [comment]
@LakshmiKumar23 

```

CMake Error at cmake/FindMIVisionX.cmake:88 (file):
  file failed to open for reading (No such file or directory):

    /opt/rocm/include/mivisionx/vx_ext_rpp_version.h
Call Stack (most recent call first):
  rocAL/CMakeLists.txt:35 (find_package)


CMake Error at cmake/FindMIVisionX.cmake:89 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  rocAL/CMakeLists.txt:35 (find_package)


CMake Error at cmake/FindMIVisionX.cmake:91 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  rocAL/CMakeLists.txt:35 (find_package)


CMake Error at cmake/FindMIVisionX.cmake:93 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  rocAL/CMakeLists.txt:35 (find_package)


```

**Comment by kiritigowda** [comment]
http://math-ci.amd.com/blue/organizations/jenkins/main%2Fprecheckin%2FrocAL/detail/PR-156/1/pipeline

**Comment by LakshmiKumar23** [comment]
> /opt/rocm/include/mivisionx/vx_ext_rpp_version.h

@kiritigowda Need to use the latest mivisionx to get this header file that I added. `   /opt/rocm/include/mivisionx/vx_ext_rpp_version.h`

**Comment by kiritigowda** [comment]
> > /opt/rocm/include/mivisionx/vx_ext_rpp_version.h
> 
> @kiritigowda Need to use the latest mivisionx to get this header file that I added. The versioning header is not available on master. `/opt/rocm/include/mivisionx/vx_ext_rpp_version.h`

You need to check if the file exists? if not, do nothing

**Comment by LakshmiKumar23** [comment]
> > > /opt/rocm/include/mivisionx/vx_ext_rpp_version.h
> > 
> > 
> > @kiritigowda Need to use the latest mivisionx to get this header file that I added. The versioning header is not available on master. `/opt/rocm/include/mivisionx/vx_ext_rpp_version.h`
> 
> You need to check if the file exists? if not, do nothing

@kiritigowda checking for file, else setting to 0

**Comment by kiritigowda** [comment]
@LakshmiKumar23 - PR failing on CI - http://math-ci.amd.com/blue/organizations/jenkins/main%2Fprecheckin%2FrocAL/detail/PR-156/2/pipeline/246

```

[ 38%] Building CXX object rocAL/CMakeFiles/rocal.dir/source/decoders/video/ffmpeg_video_decoder.cpp.o
In file included from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/decoders/audio/generic_audio_decoder.h:25,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/source/decoders/audio/generic_audio_decoder.cpp:23:
/var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/decoders/audio/audio_decoder.h:29:10: fatal error: sndfile.h: No such file or directory
   29 | #include "sndfile.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[2]: *** [rocAL/CMakeFiles/rocal.dir/build.make:668: rocAL/CMakeFiles/rocal.dir/source/decoders/audio/generic_audio_decoder.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_read_and_decode.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader.h:29,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader_sharded.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/node_audio_loader.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/master_graph.h:41,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/context.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/source/api/rocal_api_data_transfer.cpp:24:
/var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/decoders/audio/audio_decoder.h:29:10: fatal error: sndfile.h: No such file or directory
   29 | #include "sndfile.h"
      |          ^~~~~~~~~~~
In file included from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_read_and_decode.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader.h:29,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader_sharded.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/node_audio_loader.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/master_graph.h:41,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/context.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/source/api/rocal_api_info.cpp:24:
/var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/decoders/audio/audio_decoder.h:29:10: fatal error: sndfile.h: No such file or directory
   29 | #include "sndfile.h"
      |          ^~~~~~~~~~~
compilation terminated.
compilation terminated.
In file included from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_read_and_decode.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader.h:29,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader_sharded.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/node_audio_loader.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/master_graph.h:41,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/context.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/source/api/rocal_api_meta_data.cpp:28:
/var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/decoders/audio/audio_decoder.h:29:10: fatal error: sndfile.h: No such file or directory
   29 | #include "sndfile.h"
      |          ^~~~~~~~~~~
compilation terminated.
In file included from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_read_and_decode.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader.h:29,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader_sharded.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/node_audio_loader.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/master_graph.h:41,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/context.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/source/api/rocal_api.cpp:29:
/var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/decoders/audio/audio_decoder.h:29:10: fatal error: sndfile.h: No such file or directory
   29 | #include "sndfile.h"
      |          ^~~~~~~~~~~
compilation terminated.
In file included from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_read_and_decode.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader.h:29,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader_sharded.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/node_audio_loader.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/master_graph.h:41,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/context.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/source/api/rocal_api_data_loaders.cpp:29:
/var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/decoders/audio/audio_decoder.h:29:10: fatal error: sndfile.h: No such file or directory
   29 | #include "sndfile.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[2]: *** [rocAL/CMakeFiles/rocal.dir/build.make:150: rocAL/CMakeFiles/rocal.dir/source/api/rocal_api_data_transfer.cpp.o] Error 1
make[2]: *** [rocAL/CMakeFiles/rocal.dir/build.make:164: rocAL/CMakeFiles/rocal.dir/source/api/rocal_api_info.cpp.o] Error 1
In file included from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_read_and_decode.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader.h:29,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/audio_loader_sharded.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/loaders/audio/node_audio_loader.h:24,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/master_graph.h:41,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/pipeline/context.h:27,
                 from /var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/source/api/rocal_api_augmentation.cpp:26:
/var/jenkins_home/workspace/main_precheckin_rocAL_PR-156/vNVxeitcL/rocal/rocAL/include/decoders/audio/audio_decoder.h:29:10: fatal error: sndfile.h: No such file or directory
   29 | #include "sndfile.h"

```

File: rocAL/CMakeLists.txt [hunk 1]
```diff
@@ -293,6 +293,18 @@ if(${BUILD_ROCAL})
     else()
         message(FATAL_ERROR "No filesystem library found.")
     endif()
+    
+    # Audio features for rocAL
+    if(${VX_EXT_RPP_VERSION_MAJOR} VERSION_GREATER_EQUAL "3" AND ${VX_EXT_RPP_VERSION_MINOR} VERSION_GREATER_EQUAL "0" AND ${VX_EXT_RPP_VERSION_PATCH} VERSION_GREATER "0")
+        set(AUDIO_SUPPORT TRUE)
```
Comment (kiritigowda) [inline rocAL/CMakeLists.txt:299]: Also needs the deps to be installed

Comment (LakshmiKumar23) [reply rocAL/CMakeLists.txt:299]: It already checks for sndfile in line 308. Let me check why it doesn't check that if statement

Comment (LakshmiKumar23) [reply rocAL/CMakeLists.txt:299]: @kiritigowda pushed a change. Is CI down?
+ docker build -t compute-rocm-dkms-no-npi-hipclang-13897/ubuntu22gfx90a/math-ci:step1 --pull -f dockerfile-build-ubuntu22 --build-arg library_dependencies= --build-arg user_uid=1152 --build-arg base_image=10.216.51.87:5000/rocm-plus-docker/compute-rocm-dkms-no-npi-hipclang:13897-ubuntu-22.04-stg1 --build-arg build_num=13897 --cpuset-cpus=48,112,49,113,50,114,51,115,52,116,53,117,54,118,55,119,56,120,57,121,58,122,59,123,60,124,61,125,62,126,63,127 .

DEPRECATED: The legacy builder is deprecated and will be removed in a future release.

            Install the buildx component to build images with BuildKit:

            https://docs.docker.com/go/buildx/



Sending build context to Docker daemon  80.26MB


Step 1/15 : ARG base_image

Step 2/15 : FROM ${base_image}

error parsing HTTP 400 response body: invalid character '<' looking for beginning of value: "<html>\r\n<head><title>400 The plain HTTP request was sent to HTTPS port</title></head>\r\n<body>\r\n<center><h1>400 Bad Request</h1></center>\r\n<center>The plain HTTP request was sent to HTTPS port</center>\r\n<hr><center>nginx/1.23.4</center>\r\n</body>\r\n</html>\r\n"

script returned exit code 1

File: CMakeLists.txt [hunk 2]
```diff
@@ -121,7 +120,6 @@ message("-- ${BoldBlue}rocAL Build Type -- ${CMAKE_BUILD_TYPE}${ColourReset}")
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 list(APPEND CMAKE_PREFIX_PATH ${ROCM_PATH} ${ROCM_PATH}/hip)
 find_package(HALF QUIET)
-find_package(HIP QUIET)
```
Comment (LakshmiKumar23) [inline CMakeLists.txt:123]: @kiritigowda @SundarRajan28  Removed this line since it's not need at top level and causes the HIP issue noticed by Sundar

**Review by rrawther — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (SundarRajan28) [reply CMakeLists.txt:123]: This should work. I will close PR#156

**Comment by kiritigowda** [comment]
@LakshmiKumar23 merge pending CI - http://math-ci.amd.com/blue/organizations/jenkins/main%2Fprecheckin%2FrocAL/detail/PR-156/9/pipeline

**Pull request merged**